client/src/main/java/org/apache/celeborn/client/read/WorkerPartitionReader.java

```diff
@@ -156,13 +173,21 @@ public boolean hasNext() {
     return returnedChunks < endChunkIndex - startChunkIndex + 1;
   }
 
+  private void checkpoint() {
+    if (lastReturnedChunkId != -1) {
+      partitionReaderCheckpointMetadata.ifPresent(
+          readerCheckpointMetadata -> readerCheckpointMetadata.checkpoint(lastReturnedChunkId));
+    }
+  }
+
   @Override
   public ByteBuf next() throws IOException, InterruptedException {
     checkException();
+    checkpoint();
```

Comment:
nit: IMO `checkpoint` can be placed before `checkException`

Reply:
Yes makes sense. Ack